lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -184,7 +184,7 @@ def index_all(indexable_paths: RubyIndexer.configuration.indexables, &block)
 
     sig { params(indexable_path: IndexablePath, source: T.nilable(String)).void }
     def index_single(indexable_path, source = nil)
-      content = source || File.read(indexable_path.full_path)
+      content = source || File.read(indexable_path.full_path).force_encoding("UTF-8")
```

Comment:
I somehow remember @nirvdrum advocating not to use this method in his 2023 RubyKaigi talk 😂

Reply:
Yeah, I'm not sure this is the right solution.

Why isn't the file encoding `UTF-8` when we read it? Is the problem that the file is using an incompatible encoding?

And if we force everything to be `UTF-8`, can we find ourselves in a situation where we're converting incompatible data into the encoding? Maybe the solution would be to rescue, warn and ignore files with encoding issues.

In other words, it's not clear to me if this is a mistake in the gem that we should be resilient against or if this is something we need to enforce from our side.

@nirvdrum and @kddnewton do you have thoughts on this one?